0000-catalog.md

```diff
@@ -0,0 +1,789 @@
+
+```
+CatalogProductCreate: Agent from org needs the "can_create_product" permission
+CatalogProductUpdate: Agent from org needs the "can_update_product" permission
+CatalogProductDelete: Agent from org needs the "can_delete_product" permission
+```
+
+In the case of **catalog operations** agents from the org they represent will
+need the operational permission to perform the desired operation on a Grid
+Catalog.
+
+```
+CatalogProductSetStatus: Agent from org needs the "can_set_catalog_product_status" 
+permission
+```
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## State
+
+### Catalog Representation
+
+The primary object stored in state is `Catalog`, which consists of a
+`catalog_id` (hash of the catalog_name), an `owner` (org_id
+compatible w/Pike), a `name`, and a repeated field of `PropertyValues`. 
+The properties available are defined by the Grid Property Schema Transaction 
+Family and are restricted to the fields and rules of the GS1 Catalog schema 
+(which will be defined at a later time).  Transactions which are responsible 
+for setting catalog state values must ensure that those properties conform with 
+the requirements of the GS1 Catalog Schema (not yet defined).
+
+```
+message Catalog {
+    string catalog_id = 1;
+    string owner = 2;
+    string name = 3;
+    repeated PropertyValue properties = 4;
+}
+```
+
+### Referencing a Catalog
+
+Catalogs are uniquely referenced by their catalog_id. For example:
+
+```
+get_catalog(org_id, catalog_id) // GS1 Company Prefix, hash(catalog_name)
+set_catalog(org_id, catalog_id, catalog) // GS1 Company Prefix, hash(catalog_name), catalog
+```
+
+### Catalog Addressing in the Merkle-Radix State System
+
+In order to uniquely locate Catalogs in the Merkle-Radix state system, an
+address must be constructed which identifies the storage location of the Catalog
+representation.
+
+All Grid addresses are prefixed by the 6-hex-character namespace prefix
+"621dee", Catalogs are further prefixed under the Grid namespace with reserved
+enumerations of "03" ("00", "01", and "02" being reserved for other purposes)
+indicating "Grid Catalogs" and an additional org_id. Currently in Grid Product,
+org_ids are a [6-10 digit GS1 company prefix](https://www.gs1-us.info/gs1-company-prefix/).
+To keep the prefix length consistent, any GS1 company prefix less than 10
+digits will be left padded with the hexadecimal representation of the number 10 
+("a") until it is 10 digits in length. This composite address will enable 
+inexpensive catalog operations.
+
+Therefore, all addresses starting with:
+
+```
+"621dee" + "03" + "aaaa123456" // org_id (in this case a gs1_company_prefix)
+```
+
+are Grid GS1 Catalogs identified by the hash of the catalog name which can be
+referenced by a "catalog_product" to group products to a specific catalog.
+
+The catalog_id format consists of 70-digit "alphanumeric string" which include
```

Comment:
s/consists of 70-digit/consists of a 70-digit/

Also, this should probably read 70-character, instead of 70-digit.

Reply:
✅